tests/main/snap-quota-journal/task.yaml

```diff
@@ -0,0 +1,96 @@
+  # correctly for journald
+  echo "Create a group with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-journal-quota
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services test-snapd-journal-quota.logger | MATCH "test-snapd-journal-quota.logger\s+enabled\s+active"
+
+  # Verify the contents of /etc/systemd/journald@snap-group-one.conf
+  echo "Verifying the journal config contents"
+  MATCH "^SystemMaxUse=16000000"        < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RuntimeMaxUse=16000000"       < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitIntervalSec=10000us" < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitBurst=100"           < /etc/systemd/journald@snap-group-one.conf
+  
+  # Now ask for the namespace logs and lets see if we can see messages from 'LogDaemon'
+  retry -n 5 sh -c "$TESTSTOOLS/journal-state get-log --namespace snap-group-one | MATCH LogDaemon"
+
+  # Change the size and ensure that the journal is restarted
+  echo "Updating size of journal to ensure size changes"
+  snap set-quota group-one --journal-size=64MB
+
+  # Ensure that the new size is printed in the journal logs. We requested 64MB but
+  # this will most likely print out as 61.0M, so lets be a bit flexible
+  "$TESTSTOOLS"/journal-state get-log --namespace snap-group-one | MATCH "max 6[0-9].[0-9]M,"
+
+  # Ensure that the unit file is correct
+  echo "Ensuring the unit file is correct"
+  systemctl show --property=LogNamespace snap.test-snapd-journal-quota.logger.service | MATCH "LogNamespace=snap-group-one"
+  LOGGER_PID=$(systemctl show --property=MainPID --value snap.test-snapd-journal-quota.logger.service)
+
+  echo "Removing the quota will stop the slice and the service will be restarted"
+  snap remove-quota group-one
+  systemctl show --property=MainPID --value snap.test-snapd-journal-quota.logger.service | NOMATCH "$LOGGER_PID"
+  snap services test-snapd-journal-quota.logger | MATCH "test-snapd-journal-quota.logger\s+enabled\s+active"
+
+  echo "And the service is not in a slice anymore"
+  systemctl show --property=ControlGroup snap.test-snapd-journal-quota.logger.service | NOMATCH "/$sliceName/snap.test-snapd-journal-quota.logger.service"
+
+  echo "And the slice is not active anymore"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=inactive"
+
+  # Make sure that the service is running still and now outputting logs
+  # into the original journal.
+  echo "And the service is running and is outputting logs into the original journal"
+  INITIAL_NO=$("$TESTSTOOLS"/journal-state get-log | grep "LogDaemon" | tail -n 1 | awk '{print $NF}')
+  for i in $(seq 5); do
+      sleep 1
```

Comment:
I'd change this to `0.1`

Reply:
I changed it to 0.2s now but the code will still wait for 5s because this could run on really slow VMs and too small timeouts tend to break in these environments.